src/main/java/codeu/model/data/PrivateConversation.java

```diff
@@ -0,0 +1,99 @@
+/* PrivateConversation.java */
+/* Supports functionality to create private conversations, add users, and schedule 
+   surprises that correspond to adding another user in at a specified time */
+
+package codeu.model.data;
+
+import java.time.Instant; 
+import java.util.UUID; 
+import java.util.HashMap; 
+import java.util.ArrayList;
+import java.util.Timer; 
+import java.util.TimerTask;  
+import java.util.Date; 
+import java.util.Calendar; 
+
+/* Class representing a conversation, which can be thought of as a private chat room, created 
+   by a User, contains allowed Users, messages, and  future users to be added */
+public class PrivateConversation extends Conversation {
+
+    /* private class for tasks that send a notification and adds user to this conversation at scheduled time */
+    private class SurpriseTask extends TimerTask {
+        /* ID of user to be surprised */
+        private UUID id; 
+
+        public SurpriseTask(UUID id) {
+            this.id = id; 
+        }
+
+        public void run() {
+            System.out.println("User" + id + " surprised!"); // change to send alert to notification class 
+            existingUsers.put(id, true); 
+            scheduledSurprises.remove(id); 
+            timer.cancel(); 
+        }
+    }
+
+    /// timer to keep track of TimerTasks
+    private Timer timer; 
+
+    // maps UUID to time user is scheduled to be surprised 
+    public HashMap<UUID, SurpriseTask> scheduledSurprises;  
```

Comment:
Is this so that multiple people can be surprised?

Reply:
Yes! Technically not needed in this class by itself, but I imagine it might be useful if we need to easily access the times in the future.